### ISSUE TITLE
The future of leaflet, roadmap, near future plans

### ISSUE THREAD

**Issue by jcimoch** [Leaflet/Leaflet#6882]
Hi, I'm not sure if that kind of question should be asked here but I would like to get answer from core developers. 
To give you a little bit background to my question, I'm currently working on a big project heavily relaying on leaflet and many plugins or custom solutions extending it's capabilities. But as requirements changes I feel that more and more I'm fighting with leaflet, so I started looking into solutions like open layers or cesium (however 2d view is still must have so cesium is less an alternative). However I don't see alternative that would be better at every angle right now. So I started to wonder what the future holds for leaflet? 
1. Are there any plans for major refactor like typescript rewrite?
2. Are there any plans for better integration with canvas or webgl? current canvas renderer is limited to some subset of the features and also doesn't allow to easily change the order of rendering items dynamically, it would be nice to have a pane within canvas (ofcourse not as dom element but more like virtual layer so the order could be easily managed) and many more. I know there are plugins for webgl/canvas but I'm in favour of having such things as part of a core, in my opinion plugins should add small functionalities on top cause the big ones rarly work smoothly.
3. Is this project only community driven at the moment? Are there any sponsors or ways to support project (apart from opening PR) 
4. If there are any plans of major refactor what about current plugins, is some legacy support considered?

**Comment by mullinz** [comment]
> Hi, I'm not sure if that kind of question should be asked here but I would like to get answer from core developers.
> To give you a little bit background to my question, I'm currently working on a big project heavily relaying on leaflet and many plugins or custom solutions extending it's capabilities. But as requirements changes I feel that more and more I'm fighting with leaflet, so I started looking into solutions like open layers or cesium (however 2d view is still must have so cesium is less an alternative). However I don't see alternative that would be better at every angle right now. So I started to wonder what the future holds for leaflet?
> 
> 1. Are there any plans for major refactor like typescript rewrite?
> 2. Are there any plans for better integration with canvas or webgl? current canvas renderer is limited to some subset of the features and also doesn't allow to easily change the order of rendering items dynamically, it would be nice to have a pane within canvas (ofcourse not as dom element but more like virtual layer so the order could be easily managed) and many more. I know there are plugins for webgl/canvas but I'm in favour of having such things as part of a core, in my opinion plugins should add small functionalities on top cause the big ones rarly work smoothly.
> 3. Is this project only community driven at the moment? Are there any sponsors or ways to support project (apart from opening PR)
> 4. If there are any plans of major refactor what about current plugins, is some legacy support considered?

Not sure this helps, but cesium does have [2D scene support](https://cesium.com/docs/cesiumjs-ref-doc/SceneMode.html).

**Comment by jcimoch** [comment]
> 
> 
> > Hi, I'm not sure if that kind of question should be asked here but I would like to get answer from core developers.
> > To give you a little bit background to my question, I'm currently working on a big project heavily relaying on leaflet and many plugins or custom solutions extending it's capabilities. But as requirements changes I feel that more and more I'm fighting with leaflet, so I started looking into solutions like open layers or cesium (however 2d view is still must have so cesium is less an alternative). However I don't see alternative that would be better at every angle right now. So I started to wonder what the future holds for leaflet?
> > 
> > 1. Are there any plans for major refactor like typescript rewrite?
> > 2. Are there any plans for better integration with canvas or webgl? current canvas renderer is limited to some subset of the features and also doesn't allow to easily change the order of rendering items dynamically, it would be nice to have a pane within canvas (ofcourse not as dom element but more like virtual layer so the order could be easily managed) and many more. I know there are plugins for webgl/canvas but I'm in favour of having such things as part of a core, in my opinion plugins should add small functionalities on top cause the big ones rarly work smoothly.
> > 3. Is this project only community driven at the moment? Are there any sponsors or ways to support project (apart from opening PR)
> > 4. If there are any plans of major refactor what about current plugins, is some legacy support considered?
> 
> Not sure this helps, but cesium does have [2D scene support](https://cesium.com/docs/cesiumjs-ref-doc/SceneMode.html).

I'm not sure if the same kind of look/feel with mercator projection like in leaflet or openlayers i possible? because from what I could see is more like a flat plane with applied globe texture rather than full fledged 2d map view.

**Comment by mullinz** [comment]
Yes I believe it renders a 4326 projection, not familiar enough with it to know if you can modify that.

**Comment by cherniavskii** [comment]
Hi @jcimoch 
Thanks for your questions.
While I do not represent all Leaflet maintainers team, here are my thoughts:

> 1. Are there any plans for major refactor like typescript rewrite?

No.

> 2. Are there any plans for better integration with canvas or webgl? current canvas renderer is limited to some subset of the features and also doesn't allow to easily change the order of rendering items dynamically, it would be nice to have a pane within canvas (ofcourse not as dom element but more like virtual layer so the order could be easily managed) and many more. I know there are plugins for webgl/canvas but I'm in favour of having such things as part of a core, in my opinion plugins should add small functionalities on top cause the big ones rarly work smoothly.

As you mentioned, there are plugins for webgl/canvas and we're not going to include such functionality in Leaflet core.

> 3. Is this project only community driven at the moment? Are there any sponsors or ways to support project (apart from opening PR)

Yes, project is only community driven, and AFAIK we don't have any sponsors. If you're willing to sponsor Leaflet development - let us know.

> 4. If there are any plans of major refactor what about current plugins, is some legacy support considered?

Unfortunately, we have very limited resources for Leaflet development - so we do not plan refactors.

**Comment by wisdomware** [comment]
> > > Hi, I'm not sure if that kind of question should be asked here but I would like to get answer from core developers.
> > > To give you a little bit background to my question, I'm currently working on a big project heavily relaying on leaflet and many plugins or custom solutions extending it's capabilities. But as requirements changes I feel that more and more I'm fighting with leaflet, so I started looking into solutions like open layers or cesium (however 2d view is still must have so cesium is less an alternative). However I don't see alternative that would be better at every angle right now. So I started to wonder what the future holds for leaflet?
> > > 
> > > 1. Are there any plans for major refactor like typescript rewrite?
> > > 2. Are there any plans for better integration with canvas or webgl? current canvas renderer is limited to some subset of the features and also doesn't allow to easily change the order of rendering items dynamically, it would be nice to have a pane within canvas (ofcourse not as dom element but more like virtual layer so the order could be easily managed) and many more. I know there are plugins for webgl/canvas but I'm in favour of having such things as part of a core, in my opinion plugins should add small functionalities on top cause the big ones rarly work smoothly.
> > > 3. Is this project only community driven at the moment? Are there any sponsors or ways to support project (apart from opening PR)
> > > 4. If there are any plans of major refactor what about current plugins, is some legacy support considered?
> > 
> > 
> > Not sure this helps, but cesium does have [2D scene support](https://cesium.com/docs/cesiumjs-ref-doc/SceneMode.html).
> 
> I'm not sure if the same kind of look/feel with mercator projection like in leaflet or openlayers i possible? because from what I could see is more like a flat plane with applied globe texture rather than full fledged 2d map view.

I used to have my maps created in Mapbox JS. However, they change to version 2 and start charging money. I did some research and found Cesium. I rewrote all my maps in Cesium and never looked back. 

Cesium supports 2D, 3D, and Columbus views. It's open-source and has a large community following.

**Comment by jcimoch** [comment]
@wisdomware did you manage to achive same look & feel of 2D map in cesium? I find their 2D view just a different camera projection over 3D space view, it doesn't behave as you would expect from 2D map like leaflet or mapbox, or am I missing something? I was just looking over examples on their site, have no experience in cesium.
